src/pipeline/node/DynamicCalibration.cpp

```diff
@@ -0,0 +1,529 @@
+
+    std::shared_ptr<dcl::CameraCalibrationHandle> calibA =
+        createDCLCameraCalibration(calibData.leftCameraMatrix, calibData.leftDistortionCoefficients, calibData.rotationMatrixA, calibData.translationVectorA);
+    std::shared_ptr<dcl::CameraCalibrationHandle> calibB =
+        createDCLCameraCalibration(calibData.rightCameraMatrix, calibData.rightDistortionCoefficients, calibData.rotationMatrixB, calibData.translationVectorB);
+
+    const dcl::resolution_t resolution = {
+        .width = static_cast<unsigned int>(width),
+        .height = static_cast<unsigned int>(height)
+    };
+
+    sensorA = std::make_shared<dcl::CameraSensorHandle>(calibA, resolution);
+    sensorB = std::make_shared<dcl::CameraSensorHandle>(calibB, resolution);
+
+    socketA = static_cast<dcl::socket_t>(boardSocketA);
+    socketB = static_cast<dcl::socket_t>(boardSocketB);
+
+    dynCalibImpl->addSensor(deviceName, sensorA, socketA);
+    dynCalibImpl->addSensor(deviceName, sensorB, socketB);
+}
+
+void DynamicCalibration::resetResults(){
+    calibQuality.calibrationQuality = dcl::CalibrationData{};
+    calibQuality.coverageQuality = dcl::CoverageData{};
+    calibQuality.dataAquired = 0.f;
+    dynResult.calibOverallQuality = dai::DynamicCalibrationResults::CalibrationQualityResult::fromDCL(calibQuality);
+    dynResult.newCalibration = DynamicCalibrationResults::CalibrationResult::Invalid();
+    dynResult.calibOverallQuality = DynamicCalibrationResults::CalibrationQualityResult::Invalid();
+    dynResult.newCalibration = DynamicCalibrationResults::CalibrationResult::Invalid();
+    dynResult.info = "";
+};
+
+bool isAlmostBlackOrWhite(const cv::Mat& frame, double tolerance = 25.5, double maxStdDev = 10.0)
+{
+    if (frame.empty()) return true;
+
+    cv::Mat gray;
+    if (frame.channels() == 3) {
+        cv::cvtColor(frame, gray, cv::COLOR_BGR2GRAY);
+    } else {
+        gray = frame;
+    }
+
+    cv::Scalar mean, stddev;
+    cv::meanStdDev(gray, mean, stddev);
+
+    double meanVal = mean[0];
+    double stdVal = stddev[0];
+    std::cout << meanVal << " " << stdVal << std::endl;
+    bool isAlmostBlack = (meanVal < tolerance && stdVal < maxStdDev);
+    bool isAlmostWhite = (meanVal > (255 - tolerance) && stdVal < maxStdDev);
+
+    return isAlmostBlack || isAlmostWhite;
+}
+
+void DynamicCalibration::run() {
+    if(!device) {
+        logger::error("Dynamic calibration node has to have access to a device!");
+        return;
+    }
+
+    logger::info("DynamicCalibration node is running");
+    auto lastAutoTrigger = std::chrono::steady_clock::now() - std::chrono::seconds(10);
+    auto continiousTrigger = std::chrono::steady_clock::now() - std::chrono::seconds(10);
+    resetResults();
+    while(isRunning()) {
+        // auto leftFrame = left.get<dai::ImgFrame>();
+        // auto rightFrame = right.get<dai::ImgFrame>();
+        auto inSyncGroup = inSync.tryGet<dai::MessageGroup>();
```

Comment:
tryGet followed by a if(expr) continue;
has the potential to spin the CPU ...

Why is this logic needed?

What about
```suggestion
        auto inSyncGroup = inSync.get<dai::MessageGroup>();
```

and remove `if (!inSyncGroup) continue;`

Reply:
REmoved